pygfx/utils/show.py

```diff
@@ -11,56 +15,229 @@
+            scene = object
+        else:
+            scene = Scene()
+            scene.add(object)
 
-    def check_light(ob):
-        nonlocal light_count
-        if isinstance(ob, Light):
-            light_count += 1
+            dark_gray = np.array((169, 167, 168, 255)) / 255
+            light_gray = np.array((100, 100, 100, 255)) / 255
 
-    scene.traverse(check_light, False)
-    if not light_count:
-        camera.add(DirectionalLight(0.8))
-        scene.add(AmbientLight(0.2))
+            background = Background(None, BackgroundMaterial(light_gray, dark_gray))
+            scene.add(background)
+            scene.add(AmbientLight(), DirectionalLight())
+        self.scene = scene
 
-    canvas = WgpuCanvas()
-    renderer = WgpuRenderer(canvas)
+        if not any(self.find_children(Light)):
+            warnings.warn(
+                "Your scene does not contain any lights. Some objects may not be visible"
+            )
 
-    controller = OrbitController(camera.position.clone(), look_at, up=up)
-    controller.add_default_event_handlers(renderer, camera)
+        existing_cameras = self.find_children(Camera)
+        if self.camera:
+            pass
+        elif any(existing_cameras):
+            self.camera = existing_cameras[0]
+        elif not self.camera:
+            self.camera = PerspectiveCamera(70, 16 / 9)
+            self.camera.add(DirectionalLight())
+            self.scene.add(self.camera)
 
-    def animate():
-        controller.update_camera(camera)
-        renderer.render(scene, camera)
+        if self.renderer is None and self.canvas is None:
+            from wgpu.gui.auto import WgpuCanvas
+
+            self.canvas = WgpuCanvas()
+            self.renderer = WgpuRenderer(self.canvas)
+        elif self.renderer is not None:
+            self.canvas = self.renderer.target
+        elif self.canvas is not None:
+            self.renderer = WgpuRenderer(self.canvas)
+        elif self.canvas != self.renderer.target:
+            raise ValueError("Display's render target differs from it's canvas.")
+        else:
+            pass
+
+        if self.controller is None:
+            look_at = self.camera.show_object(object)
+            self.controller = OrbitController(
+                self.camera.position.clone(), look_at, up=up
+            )
+            self.controller.add_default_event_handlers(self.renderer, self.camera)
+
+        self.canvas.request_draw(self.draw_function)
+        sys.modules[self.canvas.__module__].run()
+
+    # this should probably live on WorldObject
+    def find_children(self, clazz):
+        """Return all children of the given type"""
+        objects = list()
+
+        # can we make traverse a generator?
+        # [x for x in self.scene.traverse(filter=lambda x: isinstance(x, clazz))
+        self.scene.traverse(
+            lambda x: objects.append(x) if isinstance(x, clazz) else None
+        )
+
+        return objects
+
+
+def show(
```

Comment:
I wonder if whether we'd ever use this, since we've found that we need that `display` object to be able to pick it up in the gallery and tests ...

Reply:
Still nice to have I guess, and does not look like a maintenance burden, so let's just keep it.